packages/perseus-editor/src/util/issue-ctas-utils.ts

```diff
@@ -0,0 +1,138 @@
+import {PerseusMarkdown, Util} from "@khanacademy/perseus";
+import {
+    generateImageOptions,
+    generateImageWidget,
+    type PerseusRenderer,
+    type PerseusWidgetsMap,
+} from "@khanacademy/perseus-core";
+
+// Match image markdown syntax: ![alt text](url)
+const IMAGE_MARKDOWN_REGEX = /!\[[^\]]*\]\([^)]+\)/g;
+
+type IssueCta = {
+    label: string;
+    onClick: () => void;
+};
+
+// Exported for testing
+/**
+ * Based on the given widgets within a question, return the first available
+ * index for a specified widget type.
+ *
+ * Example: If the question has two image widgets named "image 1" and "image 3",
+ * the next available index for an image widget would be 2, for "image 2".
+ *
+ * @param widgetType - The type of the widget to get the index for. (e.g. "image", "radio", etc.)
+ * @param widgets - The question's `widgets` to search through.
+ * @returns The first available index for the given widget type, starting from 1.
+ */
+export function getFirstAvailableWidgetIndex(
+    widgetType: string,
+    widgets: PerseusWidgetsMap,
+): number {
+    const widgetNames = Object.keys(widgets);
+
+    for (let i = 1; i <= widgetNames.length + 1; i++) {
+        if (!widgetNames.includes(`${widgetType} ${i}`)) {
+            return i;
+        }
+    }
+
+    return 1;
+}
+
+/**
+ * Convert all image markdown in a question to image widgets.
+ *
+ * This function uses regex to find image markdown, then replaces it with an
+ * image widget. It does not use PerseusMarkdown to parse the content for image
+ * markdown, because PerseusMarkdown doesn't have a node-to-text converter. We'd
+ * have to manually process the tree to convert every single other markdown syntax
+ * type back to its text content form.
+ *
+ * @param question - The question being parsed for image markdown.
+ * @param onEditorChange - The function to update the editor's content.
+ */
+export async function convertImageMarkdownToImageWidget(
+    question: PerseusRenderer,
+    onEditorChange: (newProps: any) => void,
+) {
+    const {content, widgets} = question;
+    const newWidgets = {...widgets};
+
+    // Find all image markdown matches
+    const matches = Array.from(content.matchAll(IMAGE_MARKDOWN_REGEX));
```

Comment:
We should not be inspecting the `content` using regexes.  Instead we should use Perseus' Markdown-parsing library and search that way.  See the [FAQ](https://khanacademy.atlassian.net/wiki/spaces/ENG/pages/4300800034/Perseus+FAQ) for details.